Code/Framework/AzQtComponents/AzQtComponents/Application/Window/AzQtApplicationWindow.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+* All or portions of this file Copyright (c) Amazon.com, Inc. or its affiliates or
+* its licensors.
+*
+* For complete copyright and license terms please see the LICENSE at the root of this
+* distribution (the "License"). All use of this software is governed by the License,
+* or, if provided, by the license below or the license accompanying this file. Do not
+* remove or modify any license notices. This file is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*
+*/
+
+#include <AzQtComponents/Application/Window/AzQtApplicationWindow.h>
+#include <QVBoxLayout>
+
+
+namespace AzQtComponents
+{
+    AzQtApplicationWindow::AzQtApplicationWindow(QWidget* parent /* = 0 */, const AZStd::string& objectName)
+        : AzQtComponents::DockMainWindow(parent)
+    {
+        m_advancedDockManager = new AzQtComponents::FancyDocking(this);
```

Comment:
This will need to be given a unique identifier as the second parameter. Might make sense to move this to the actual DockMainWindow instead.

Reply:
After looking at what we've pulled out here, I'd suggest removing this class from the PR entirely - besides creating a `FancyDocking` instance, this doesn't really provide any docking specific functionality. I do think there's use for this class later though; since this is a layout that's used in a few places in Atom. I would suggest moving this to `AtomToolsFramework` as something like an `AtomToolMainWindow` class in a follow-up PR.